src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -412,6 +418,14 @@ export class ConnectionDialogService implements IConnectionDialogService {
 		}
 
 		let newProfile = new ConnectionProfile(this._capabilitiesService, model || providerName);
+		if (!model.password && defaultAuthenticationType === AuthenticationType.SqlLogin && authenticationTypeName === AuthenticationType.SqlLogin) {
```

Comment:
Is there a reason you're checking the default auth type here? If the auth type passed in was sql then I would think we'd want to fill in the password regardless of what the default type is for that provider.

Reply:
According to @cheenamalhotra, we only want to fill in the password when the default connection is SQL Login since there isn't an expectation to load the password when the auth type is switched to SQL Login.

She outlined the two things for me to take note of in this comment: https://github.com/microsoft/azuredatastudio/pull/21813#issuecomment-1414266066

Reply:
@lewis-sanchez  just above on line 413, default authenticationType is being considered so additional check is not needed.